ui/components/src/OptionsEditorLayout/OptionsEditorControl.tsx

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { FormControl, FormLabel, FormControlLabelProps, Stack, Box } from '@mui/material';
+import React from 'react';
+import { useId } from '../utils';
+
+export type OptionsEditorControlProps = Pick<FormControlLabelProps, 'label' | 'control'>;
+
+export const OptionsEditorControl = ({ label, control }: OptionsEditorControlProps) => {
```

Comment:
Would there ever be a situation where we wouldn't want to use OptionsEditorControl? It looks really flexible, but for an option like thresholds in the future, I guess we'll still be able to build that completely custom right? 
See screenshot here for ex: 

<img width="339" alt="image" src="https://user-images.githubusercontent.com/9369625/202209650-ac96523c-105d-408a-951e-03362ab27797.png">

Reply:
I think we'll need to build some additional components for more complex cases like that, but hopefully we can leverage some of the same techniques.